### ISSUE TITLE
IAM policy resource late initialization does not complete for `Description` field

### ISSUE THREAD

**Issue by vshunkov** [aws-controllers-k8s/community#1490]
**Describe the bug**
Hello team, we recognised that iam-controller does not sync properly iam-policy resource and causing constant reconciliation with message "Late initialization did not complete, requeuing with delay of 5 seconds", full log attached bellow.

**Steps to reproduce**
Deploy iam policy with iam-controller, check resource status.

**Expected outcome**
IAM policy resource state synced.

**Environment**
* EKS: 1.21
* AWS service targeted: AWS::IAM::Policy
* Controller: iam-controller:v0.0.21

[iam-controller.log](https://github.com/aws-controllers-k8s/community/files/9657109/iam-controller.log)

**Comment by A-Hilaly** [comment]
Hi @vshunkov could you provide an example of a resource that's triggering this error message?

**Comment by vshunkov** [comment]
Hi @A-Hilaly, here you are [policy.txt](https://github.com/aws-controllers-k8s/community/files/9662408/policy.txt)

**Comment by jaypipes** [comment]
Apologies on delayed response... Still digging out from hurricane...

Can you please paste the results of kubectl describe? Thanks!

**Comment by vshunkov** [comment]
Hi @jaypipes,

I hope none of your loved ones got hurt. Here is [describe output](https://github.com/aws-controllers-k8s/community/files/9708048/describe.txt).

Regards,
Vladimir.

**Comment by jaypipes** [comment]
@vshunkov OK, there's at least a couple bugs that I've found. The first is that the delta comparison for Tags is incorrectly returning a difference between lists that contain identical structs but are ordered differently. The fix for this bug will avoid the repeated lines in the controller log that look like this:

```
2022-09-27T14:37:23.536Z	INFO	ackrt	desired resource state has changed	{"account": "XXXXXXXXXX", "role": "", "region": "eu-west-1", "kind": "Policy", "namespace": "default", "name": "iamp-d-demo-test-eu-west-1", "is_adopted": false, "generation": 6, "diff": [{"Path":{"Parts":["Spec","Tags"]} ...
```

The controller isn't actually calling any AWS APIs for changing tags, but it's annoying to have the controller logs show there is a difference in tags when there actually isn't :) I will work on this bug in a separate issue.

The second bug seems to be that the fields of a Policy that are late-initialized (which are `Spec.Path` and `Spec.Description`) are not properly returning a non-error value for some reason.

I suspect that this is probably the `Spec.Description` field which is late-initialized and I wrote this in the `generator.yaml` file for the IAM controller after finding out that the `Description` field is, well, [a bit weird](https://github.com/aws-controllers-k8s/iam-controller/blob/687eb2d43958f923a3691773bae262f38e1b08cc/generator.yaml#L48-L65):

```yaml
      Description:
        # You might be wondering why description is late-initialized, since
        # there isn't a default server-side value for description.
        #
        # The CreatePolicy API call accepts a Description field in the input
        # and the documentation (and API model) say that a Description field is
        # included in the returned response from CreatePolicy, however the
        # Description returned from the CreatePolicy API call is always
        # missing/nil which means the SetResource code sets the
        # Spec.Description to nil. When the next time the GetPolicy API call is
        # made in the next reconciliation loop, GetPolicy returns the
        # description that was originally set in the CreatePolicy API call and
        # a Delta difference is discovered erroneously (because the SetResource
        # call after CreatePolicy incorrectly set the description to nil). So,
        # we set the late initialize property of the Description field here to
        # override the Spec.Description to the original value we set in the
        # CreatePolicy *input* shape.
        late_initialize: {}
```

@vshunkov May I ask that you add a `Description` to your `Policy` resource and see if that ends up allowing the late initialization to complete properly? If so, then it's a bug in how I implemented late initialization for the Description field...

**Comment by jaypipes** [comment]
OK, awesome. You can use that as a workaround in the meantime while I fix the IAM controller's late initialization of the `Description` field. I've also created a new issue (#1503) for the tag comparison bug.

**Comment by jaypipes** [comment]
Related #1772

**Comment by a-hilaly** [comment]
This is now fixed in [iam-controller v1.3.6](https://github.com/aws-controllers-k8s/iam-controller/releases/tag/v1.3.6) - the controller now correctly handles the Description field for Roles and Policies, preventing an infinite requeue caused by missing Description field in Create calls. cc @vshunkov
